fastlane_bot/helpers/txhelpers.py

```diff
@@ -969,13 +972,57 @@ def approve_token_for_arb_contract(self, token_address: str, approval_amount: in
                             base_gas_price=baseFee, max_priority_fee=max_priority, nonce=self.get_nonce()
                         )
                     )
+                    approve_tx = self.sign_transaction(approve_tx)
+                    self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
+                    return self.submit_transaction(approve_tx)
                 except Exception as e:
                     self.ConfigObj.logger.info(
                         f"(***2***) Error when building transaction: {e.__class__.__name__} {e}")
             else:
                 return None
-        self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
-        return self.submit_transaction(approve_tx)
 
+    async def get_layer_one_gas_fee(self, signed_transaction) -> Decimal:
+        """
+        Returns the expected layer one gas fee for a layer 2 Optimism transaction
```

Comment:
Is this really unique to the `Optimism` chain?

Reply:
As far as I know - yes. (Including other Optimism deployments such as Base)

Reply:
I meant to say that in the context of this module, this information is irrelevant, even if it **currently** happens to be the case.

Reply:
I think it's important to be specific here because otherwise it sounds like it could be relevant for layer one chains. Currently I'm only aware of Optimism using this model. If in the future more layer 2 solutions adopt this, we can clarify this further.

Reply:
This file is a general helper utility.
It doesn't make sense to add specific-dex code or specific-chain code in here.
Same goes for documentation.
If this function is specific to Optimism, then it should move to the Optimism module.